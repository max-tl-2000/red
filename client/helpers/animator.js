/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import $ from 'jquery';

let counter = 0;

const generateId = name => {
  counter++;
  return `${name}_${counter}`;
};

const playAnimation = (ele, anim, options) => {
  const id = generateId('animId');
  const $ele = $(ele);

  const opts = {
    timeout: 3000, // ensure the promise is resolved after 3000ms
    selector: null, // if selector is added we will use delegation
    ...options,
  };

  let timer;

  const p = new Promise(resolve => {
    const doResolve = () => {
      $ele.removeClass(anim);
      clearTimeout(timer);
      resolve();
    };

    if (opts.selector) {
      $ele.one(`animationend.${id}`, opts.selector, doResolve);
    } else {
      $ele.one(`animationend.${id}`, doResolve);
    }

    timer = setTimeout(doResolve, opts.timeout);

    $ele.addClass(anim);
  });

  return p;
};

export const create = ({ showClass = 'on', enterClass = 'enter', exitClass = 'exit' } = {}) => ({
  show(ele) {
    const $ele = $(ele);
    if ($ele.hasClass(showClass)) {
      return Promise.resolve();
    }

    $ele.addClass(showClass);
    return playAnimation($ele, enterClass);
  },
  hide(ele) {
    const $ele = $(ele);
    if (!$ele.hasClass(showClass)) {
      return Promise.resolve();
    }
    return playAnimation($ele, exitClass).then(() => $ele.removeClass(showClass));
  },
});
