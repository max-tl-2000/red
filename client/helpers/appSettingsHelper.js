/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { isValidMoment } from '../../common/helpers/moment-utils';
import { DATE_US_FORMAT } from '../../common/date-constants';

export const TRUE_STRING = true.toString();
export const FALSE_STRING = false.toString();

export const encodeTextSetting = value => {
  if (!value) return '';
  return value.trim();
};

export const encodeBoolSetting = value => {
  if (!value || value.toString() !== TRUE_STRING) return FALSE_STRING;
  return TRUE_STRING;
};

export const encodeDateSetting = value => {
  if (!isValidMoment(value)) return null;
  return value.format(DATE_US_FORMAT);
};

export const encodeNumberSetting = value => {
  if (!value) return '';
  return value.trim();
};
