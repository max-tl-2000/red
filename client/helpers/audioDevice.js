/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { loadScript } from '../../common/client/load-script';
import { window } from '../../common/helpers/globals';

const jabraOffHook = async _event => {
  window.jabra.offHook();
};

const jabraOnHook = async _event => {
  window.jabra.onHook();
};

const jabraMute = async _event => {
  window.jabra.mute();
};

const jabraUnmute = async _event => {
  window.jabra.unmute();
};

const isHandledDevice = device => device.label.toLowerCase().indexOf('jabra') > -1;

const getDevice = async () => {
  await navigator.mediaDevices
    .getUserMedia({ audio: true })
    .then(stream => {
      stream.getTracks().forEach(track => track.stop());
      console.log('Microphone access granted');
    })
    .catch(err => {
      console.warn('Microphone access not granted', err);
    });
  const devices = (await navigator.mediaDevices.enumerateDevices()).filter(device => device.deviceId !== 'default');
  console.info('Audio Device - Detected devices: ', devices);
  const jabraDevice = devices.find(isHandledDevice);
  console.info('Audio Device - Detected Jabra device: ', jabraDevice);
  return {
    deviceId: jabraDevice?.deviceId,
    handledDevice: !!jabraDevice?.deviceId,
  };
};

const registerEventListeners = async (jabra, handlers) => {
  jabra.addEventListener('reject', async event => {
    handlers?.reject && handlers.reject();
    await jabraOnHook(event);
  });

  jabra.addEventListener('acceptcall', async event => {
    handlers?.answer && handlers.answer();
    await jabraOffHook(event);
  });

  jabra.addEventListener('mute', async event => {
    handlers?.mute && handlers.mute();
    await jabraMute(event);
  });

  jabra.addEventListener('unmute', async event => {
    handlers?.unmute && handlers.unmute();
    await jabraUnmute(event);
  });

  jabra.addEventListener('endcall', async event => {
    handlers?.hangup && handlers.hangup();
    await jabraOnHook(event);
  });
};

export const initDevice = async (plivo, handlers) => {
  const connectedDevice = await getDevice();
  let jabra = window.jabra || null;

  if (connectedDevice.handledDevice && !jabra) {
    const jabraLib = '/libs/jabra/jabra.browser.integration-3.0.js';
    try {
      await loadScript(jabraLib, { async: true });
      jabra = window.jabra;
      await jabra.init();
      console.info('Audio Device - Library Initialized');
      await registerEventListeners(jabra, handlers);
    } catch (error) {
      jabra = null;
      window.jabra = null;
      console.warn('Audio Device - Library cannot be initialized', error);
    }
  }
  return { device: jabra, ...connectedDevice };
};
