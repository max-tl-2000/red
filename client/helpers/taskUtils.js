/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { t } from 'i18next';
import { DALTypes } from '../../common/enums/DALTypes';
import { formatDay } from '../../common/helpers/date-utils';
import { allowedToReviewApplication } from '../../common/acd/access';
import { toMoment, now, findLocalTimezone } from '../../common/helpers/moment-utils';
import { cronofyRsvpStatusTypes } from '../../common/enums/calendarTypes';

const taskNames = DALTypes.TaskNames;
const taskCategories = DALTypes.TaskCategories;

export const isTaskComplete = task => task.state === DALTypes.TaskStates.COMPLETED || task.state === DALTypes.TaskStates.CANCELED;

const isAppointmentMarkableAsComplete = task =>
  task.name === taskNames.APPOINTMENT && !isTaskComplete(task) && now().isAfter(toMoment(task.metadata.startDate));

export const filterVisibleTasks = (tasks, timezone) =>
  tasks.filter(task =>
    task.name === DALTypes.TaskNames.COLLECT_EMERGENCY_CONTACT && task.metadata && task.metadata.startDate
      ? now({ timezone }).isSameOrAfter(toMoment(task.metadata.startDate, { timezone }), 'day')
      : true,
  );

export const isTaskReassignable = task =>
  [taskCategories.MANUAL, taskCategories.APPOINTMENT, taskCategories.MANUAL_REMINDER].includes(task.category) ||
  [
    taskNames.FOLLOWUP_PARTY,
    taskNames.PROMOTE_APPLICATION,
    taskNames.SEND_RENEWAL_QUOTE,
    taskNames.COLLECT_EMERGENCY_CONTACT,
    taskNames.COLLECT_SERVICE_ANIMAL_DOC,
  ].includes(task.name) ||
  task.metadata?.isReassignable;

export const isTaskEditable = (task, user, party) => {
  if (isTaskComplete(task)) return false;
  if ([taskCategories.MANUAL, taskCategories.APPOINTMENT, taskCategories.MANUAL_REMINDER].includes(task.category)) {
    return true;
  }

  if (task.category === taskCategories.REQUIRE_WORK) {
    return allowedToReviewApplication(user, party);
  }

  return false;
};

const isAppointmentCreatedInThePast = appointment => toMoment(appointment.metadata.startDate).isBefore(toMoment(appointment.created_at));

const getTaskOverdueDate = task => {
  const getAppointmentOverdueDate = appointment => (isAppointmentCreatedInThePast(appointment) ? appointment.created_at : appointment.metadata.startDate);
  return task.name === taskNames.APPOINTMENT ? getAppointmentOverdueDate(task) : task.dueDate;
};

export const getTaskDueDate = task => (task.name === taskNames.APPOINTMENT ? task.metadata.startDate : task.dueDate);

export const isTaskOverdue = (task, timezone) => {
  const today = now({ timezone });
  const taskOverdueDate = getTaskOverdueDate(task);
  return toMoment(taskOverdueDate, { timezone }).isBefore(today, 'day') && !isTaskComplete(task);
};

export const getFormattedTaskTitle = (task, params = {}) => (!Object.keys(params).length ? t(task.name) : t(task.metadata?.title || task.name, { ...params }));

export const isTaskAutoclosing = task =>
  [
    taskNames.APPOINTMENT === task.name && !isAppointmentMarkableAsComplete(task) ? taskNames.APPOINTMENT : '',
    taskNames.INTRODUCE_YOURSELF,
    taskNames.REMOVE_ANONYMOUS_EMAIL,
    taskNames.CALL_BACK,
    taskNames.FOLLOWUP_PARTY,
    taskNames.COMPLETE_CONTACT_INFO,
    taskNames.REVIEW_APPLICATION,
    taskNames.PROMOTE_APPLICATION,
    taskNames.COUNTERSIGN_LEASE,
    taskNames.SEND_CONTRACT,
    taskNames.HOLD_INVENTORY,
    taskNames.SEND_RENEWAL_QUOTE,
  ].includes(task.name) && !task.metadata?.isNotAutoclosing;

export const taskDuedateFormat = (task, timezone) => {
  const { state, name, completionDate } = task;

  const taskDueDate = toMoment(getTaskDueDate(task), { timezone });
  const currentDate = now({ timezone });
  const taskCompletionDate = completionDate ? toMoment(completionDate, { timezone }) : undefined;
  const isAppointment = name === taskNames.APPOINTMENT;
  const isComplete = state === DALTypes.TaskStates.COMPLETED;

  const dateFormatForNextTasks = lowerCase => {
    const tomorrow = now({ timezone }).add(1, 'days');
    const daysDifference = taskDueDate.diff(currentDate, 'days', true);
    const dayLowerCaseFormat = day => {
      if (lowerCase) return t(day).toLowerCase();
      return t(day);
    };

    if (currentDate.isSame(taskDueDate, 'day')) {
      return dayLowerCaseFormat('DATETIME_TODAY');
    }
    if (tomorrow.isSame(taskDueDate, 'day')) {
      return dayLowerCaseFormat('DATETIME_TOMORROW');
    }
    if (daysDifference < 6) return taskDueDate.format('dddd');
    return formatDay(taskDueDate, timezone);
  };

  const envTimezone = findLocalTimezone();
  const differentZone = timezone !== envTimezone;
  const zonePart = differentZone ? ` ${taskDueDate.zoneAbbr()}` : '';

  const dateFormat = currentDate.isSame(taskDueDate, 'year') ? 'EMAIL_CARD_DATE_DEFAULT' : 'EMAIL_CARD_TIMESTAMP_WITH_YEAR';
  const dateFormatForPastAppointments = `${taskDueDate.format(t(dateFormat))} at ${taskDueDate.format('h:mma')}${zonePart}`;

  const dateFormatForNextOrSetInPastAppointments = () =>
    isAppointmentCreatedInThePast(task) ? dateFormatForPastAppointments : `${dateFormatForNextTasks()} at ${taskDueDate.format('h:mma')}${zonePart}`; // TODO I18n this

  const dateFormatForOverdueTasks = () => {
    const taskDueDateEndOfDay = taskDueDate.clone().endOf('day');
    const overdueDays = Math.ceil(currentDate.diff(taskDueDateEndOfDay, 'days', true));
    if (overdueDays === 1) return `${overdueDays} ${t('DAY')} ${t('OVERDUE')}`;
    return `${overdueDays} ${t('DAYS')} ${t('OVERDUE')}`;
  };

  const dateFormatForCompletedTasks = taskCompletionDate ? formatDay(taskCompletionDate, timezone) : '';

  const nextDateFormat = isAppointment ? dateFormatForNextOrSetInPastAppointments() : t('DUE_DATE_BY', { date: dateFormatForNextTasks(true) });
  const completedDateFormat = isAppointment ? dateFormatForPastAppointments : dateFormatForCompletedTasks;
  const overdueDateFormat = isAppointment ? dateFormatForPastAppointments : dateFormatForOverdueTasks();

  if (isComplete) return completedDateFormat;
  if (isTaskOverdue(task, timezone)) return overdueDateFormat;

  return nextDateFormat;
};

export const isTaskNameClickable = (task, currentUserId) => {
  const clickableTask =
    task.name === DALTypes.TaskNames.COUNTERSIGN_LEASE ||
    task.name === DALTypes.TaskNames.REVIEW_APPLICATION ||
    task.name === DALTypes.TaskNames.NOTIFY_CONDITIONAL_APPROVAL;
  return task.userIds.includes(currentUserId) && !isTaskComplete(task) && clickableTask;
};

export const shouldTaskShowGuestName = task => [DALTypes.TaskNames.COMPLETE_CONTACT_INFO, DALTypes.TaskNames.REMOVE_ANONYMOUS_EMAIL].includes(task.name);

export const isTaskDeclinedViaExtCalendar = task => {
  const {
    metadata: { rsvpStatuses },
  } = task;

  if (!rsvpStatuses) return false;

  const hasDeclineStatuses = rsvpStatuses.some(s => s.status === cronofyRsvpStatusTypes.DECLINED);
  if (!hasDeclineStatuses) return false;

  return true;
};

export const getFeedbackForIcsRsvpDecline = (task, partyMembers) => {
  const {
    metadata: { rsvpStatuses },
  } = task;
  const partyMemberIds = rsvpStatuses?.filter(s => s.status === cronofyRsvpStatusTypes.DECLINED).map(s => s.partyMemberId) || [];
  const personNames = partyMemberIds.length
    ? partyMembers.filter(member => partyMemberIds.includes(member.id)).map(member => member.person?.fullName || member.fullName)
    : [];

  return personNames.length ? t('DECLINED_BY', { names: personNames.join(', ') }) : '';
};

export const getTaskClosingNote = task => {
  const {
    metadata: { feedback, feedbackDate, closingNote, closingNoteDate },
  } = task;
  if (feedbackDate && closingNoteDate) return feedbackDate > closingNoteDate ? feedback : closingNote;

  return feedback || closingNote;
};

export const getTaskClosingNoteLabel = task => {
  const {
    metadata: { feedbackDate, closingNoteDate },
  } = task;
  if (feedbackDate && closingNoteDate) {
    return feedbackDate > closingNoteDate ? `${t('CANCELLED_BY_GUEST')}. ` : `${t(task.metadata.appointmentResult)}. `;
  }

  return `${feedbackDate ? t('CANCELLED_BY_GUEST') : t(task.metadata.appointmentResult)}. `;
};

export const getClosingNote = (task, partyMembers) => {
  const isAppointmentDeclined = isTaskDeclinedViaExtCalendar(task);
  const closingNoteLabel = isAppointmentDeclined ? '' : getTaskClosingNoteLabel(task);
  const closingNoteMessage = isAppointmentDeclined ? getFeedbackForIcsRsvpDecline(task, partyMembers) : getTaskClosingNote(task);

  return {
    label: closingNoteLabel,
    message: closingNoteMessage,
  };
};
