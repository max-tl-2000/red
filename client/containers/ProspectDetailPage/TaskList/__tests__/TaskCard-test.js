/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import React from 'react';
// import TaskCard from '../TaskCard';
import { mount } from 'enzyme';
import { DALTypes } from '../../../../../common/enums/DALTypes';

import * as taskUtils from '../../../../helpers/taskUtils';
import { toMoment } from '../../../../../common/helpers/moment-utils';

const { mockModules } = require('test-helpers/mocker').default(jest);

const baseDate = toMoment('2017-05-18T20:00:11').utc();

jest.resetModules();

mockModules({
  '../../../../helpers/taskUtils.js': {
    ...taskUtils,
    taskDuedateFormat: () => '1 DAY OVERDUE', // always mock this instance as 1 day overdue only
  },
});

const TaskCard = require('../TaskCard').default;

const activeIntroduceYourselfTask = {
  id: 'd5a906a9-5a9b-4e41-a40b-3a57f81d9791',
  created_at: '2016-11-23T22:30:39.232Z',
  dueDate: baseDate.format(),
  updated_at: baseDate.format(),
  name: DALTypes.TaskNames.INTRODUCE_YOURSELF,
  partyId: 'd5a906a9-5a9b-4e41-a40b-3a57f81d979a',
  state: DALTypes.TaskStates.ACTIVE,
  userIds: ['25e5702d-ef29-4996-8948-c1fc593e5b36'],
  category: 'Party',
  metadata: {},
};

const completeIntroduceYourselfTask = {
  ...activeIntroduceYourselfTask,
  state: DALTypes.TaskStates.COMPLETED,
};

const overdueIntroduceYourselfTask = {
  ...activeIntroduceYourselfTask,
  dueDate: toMoment(activeIntroduceYourselfTask.dueDate).subtract(2, 'days').format(),
};

const selectorData = {
  users: [
    {
      created_at: '2016-11-19T22:30:39.232Z',
      fullName: 'Alice Altimes',
      id: '25e5702d-ef29-4996-8948-c1fc593e5b36',
      preferredName: 'Alice',
      teams: [
        {
          id: 'd491d706-6129-4edc-a69a-ee5cd026ca68',
          displayName: 'Parkmerced Pod',
          mainRoles: ['PM'],
        },
      ],
    },
  ],
};

const party = {
  partyMembers: [{ fullName: 'Client Name 1' }, { fullName: 'Client Name 2' }],
};

const currentUser = {
  id: '0e3c5587-b27f-4e2a-bb45-9368cd2727b3',
};

describe('TaskCard', () => {
  describe('Active introduce yourself TaskCard', () => {
    const wrapper = mount(
      <TaskCard party={party} task={activeIntroduceYourselfTask} formattedValue="Introduce yourself" selectorData={selectorData} currentUser={currentUser} />,
    );
    it('should have a blank checkbox, have the text "Introduce yourself"', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Completed introduce yourself TaskCard', () => {
    const wrapper = mount(
      <TaskCard party={party} task={completeIntroduceYourselfTask} formattedValue="Introduce yourself" selectorData={selectorData} currentUser={currentUser} />,
    );

    it('should have a check icon, and task name and task date disabled', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Overdue introduce yourself TaskCard', () => {
    const wrapper = mount(
      <TaskCard party={party} task={overdueIntroduceYourselfTask} formattedValue="Introduce yourself" selectorData={selectorData} currentUser={currentUser} />,
    );

    it('should have a blank icon, use regular text to show task name and have the taskDate highlighted', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
