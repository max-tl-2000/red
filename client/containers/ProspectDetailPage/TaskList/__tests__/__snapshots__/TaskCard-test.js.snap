// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard Active introduce yourself TaskCard should have a blank checkbox, have the text "Introduce yourself" 1`] = `
<TaskCard
  currentUser={
    Object {
      "id": "0e3c5587-b27f-4e2a-bb45-9368cd2727b3",
    }
  }
  formattedValue="Introduce yourself"
  party={
    Object {
      "partyMembers": Array [
        Object {
          "fullName": "Client Name 1",
        },
        Object {
          "fullName": "Client Name 2",
        },
      ],
    }
  }
  selectorData={
    Object {
      "users": Array [
        Object {
          "created_at": "2016-11-19T22:30:39.232Z",
          "fullName": "Alice Altimes",
          "id": "25e5702d-ef29-4996-8948-c1fc593e5b36",
          "preferredName": "Alice",
          "teams": Array [
            Object {
              "displayName": "Parkmerced Pod",
              "id": "d491d706-6129-4edc-a69a-ee5cd026ca68",
              "mainRoles": Array [
                "PM",
              ],
            },
          ],
        },
      ],
    }
  }
  task={
    Object {
      "category": "Party",
      "created_at": "2016-11-23T22:30:39.232Z",
      "dueDate": "2017-05-18T20:00:11Z",
      "id": "d5a906a9-5a9b-4e41-a40b-3a57f81d9791",
      "metadata": Object {},
      "name": "INTRODUCE_YOURSELF",
      "partyId": "d5a906a9-5a9b-4e41-a40b-3a57f81d979a",
      "state": "Active",
      "updated_at": "2017-05-18T20:00:11Z",
      "userIds": Array [
        "25e5702d-ef29-4996-8948-c1fc593e5b36",
      ],
    }
  }
>
  <Row
    data-id="appointment-row"
  >
    <div
      className="row"
      data-component="row"
      data-id="appointment-row"
      style={Object {}}
    >
      <Cell
        width={70}
      >
        <div
          data-component="cell"
          style={
            Object {
              "width": 70,
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <Tooltip
                appendToBody={true}
                hoverDelay={300}
                position="top"
                text="INTRODUCE_YOURSELF_TOOLTIP"
              >
                <FlyOut
                  appendToBody={true}
                  closeOnTapAway={true}
                  expandTo="top"
                  hoverDelay={300}
                  useHover={true}
                >
                  <Icon
                    className=""
                    disabled={true}
                    name="checkbox-blank-outline"
                  >
                    <section
                      className="icon disabled"
                      data-component="icon"
                      data-red-icon="true"
                      name="checkbox-blank-outline"
                    >
                      <svg
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "<path d=\\"M19,5v14H5V5h2h3h7H19V3H5C3.9,3,3,3.9,3,5v14c0,1.1,0.9,2,2,2h14c1.1,0,2-0.9,2-2V5c0-1.1-0.9-2-2-2V5z M11,16L11,16L11,16
	L11,16L11,16L11,16L11,16z\\"></path>",
                            }
                          }
                          id="checkbox-blank-outline"
                        />
                      </svg>
                    </section>
                  </Icon>
                </FlyOut>
              </Tooltip>
            </div>
          </div>
        </div>
      </Cell>
      <Cell>
        <div
          data-component="cell"
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <div
                className="taskDetails"
                onClick={[Function]}
              >
                <text
                  disabled={false}
                >
                  <p
                    className="text"
                    data-component="text"
                    data-typography-element={true}
                  >
                    Introduce yourself
                  </p>
                </text>
                <text
                  disabled={false}
                  error={true}
                  secondary={false}
                >
                  <p
                    className="text error"
                    data-component="text"
                    data-typography-element={true}
                  >
                    1 DAY OVERDUE
                  </p>
                </text>
              </div>
            </div>
          </div>
        </div>
      </Cell>
      <Cell
        width="30%"
      >
        <div
          data-component="cell"
          style={
            Object {
              "width": "30%",
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <text
                data-id="taskOwnerName"
                disabled={false}
              >
                <p
                  className="text"
                  data-component="text"
                  data-id="taskOwnerName"
                  data-typography-element={true}
                />
              </text>
            </div>
          </div>
        </div>
      </Cell>
      <Cell
        data-id="manualTask_menu"
        textAlign="center"
        type="ctrlCell"
        width={86}
      >
        <div
          data-component="cell"
          data-id="manualTask_menu"
          style={
            Object {
              "width": 86,
            }
          }
        >
          <div
            className="cell no-padding"
            style={
              Object {
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            <div
              className="cell-inner"
              style={Object {}}
            />
          </div>
        </div>
      </Cell>
    </div>
  </Row>
</TaskCard>
`;

exports[`TaskCard Completed introduce yourself TaskCard should have a check icon, and task name and task date disabled 1`] = `
<TaskCard
  currentUser={
    Object {
      "id": "0e3c5587-b27f-4e2a-bb45-9368cd2727b3",
    }
  }
  formattedValue="Introduce yourself"
  party={
    Object {
      "partyMembers": Array [
        Object {
          "fullName": "Client Name 1",
        },
        Object {
          "fullName": "Client Name 2",
        },
      ],
    }
  }
  selectorData={
    Object {
      "users": Array [
        Object {
          "created_at": "2016-11-19T22:30:39.232Z",
          "fullName": "Alice Altimes",
          "id": "25e5702d-ef29-4996-8948-c1fc593e5b36",
          "preferredName": "Alice",
          "teams": Array [
            Object {
              "displayName": "Parkmerced Pod",
              "id": "d491d706-6129-4edc-a69a-ee5cd026ca68",
              "mainRoles": Array [
                "PM",
              ],
            },
          ],
        },
      ],
    }
  }
  task={
    Object {
      "category": "Party",
      "created_at": "2016-11-23T22:30:39.232Z",
      "dueDate": "2017-05-18T20:00:11Z",
      "id": "d5a906a9-5a9b-4e41-a40b-3a57f81d9791",
      "metadata": Object {},
      "name": "INTRODUCE_YOURSELF",
      "partyId": "d5a906a9-5a9b-4e41-a40b-3a57f81d979a",
      "state": "Completed",
      "updated_at": "2017-05-18T20:00:11Z",
      "userIds": Array [
        "25e5702d-ef29-4996-8948-c1fc593e5b36",
      ],
    }
  }
>
  <Row
    data-id="appointment-row"
  >
    <div
      className="row"
      data-component="row"
      data-id="appointment-row"
      style={Object {}}
    >
      <Cell
        width={70}
      >
        <div
          data-component="cell"
          style={
            Object {
              "width": 70,
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <Icon
                disabled={true}
                name="check"
              >
                <section
                  className="icon disabled"
                  data-component="icon"
                  data-red-icon="true"
                  name="check"
                >
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M19.6,5.6L9,16.2l-4.1-4.1l0,0L9,16.2L19.6,5.6L19.6,5.6L19.6,5.6L19.6,5.6L9,16.2l-4.1-4.1l-1.4,1.4l0,0L9,19l0,0L21,7
	L19.6,5.6L19.6,5.6z M11,16L11,16L11,16L11,16L11,16L11,16L11,16z\\"></path>",
                        }
                      }
                      id="check"
                    />
                  </svg>
                </section>
              </Icon>
            </div>
          </div>
        </div>
      </Cell>
      <Cell>
        <div
          data-component="cell"
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <div
                className="taskDetails"
                onClick={[Function]}
              >
                <text
                  disabled={true}
                >
                  <p
                    className="text disabled"
                    data-component="text"
                    data-typography-element={true}
                  >
                    Introduce yourself
                  </p>
                </text>
                <text
                  disabled={true}
                  error={false}
                  secondary={true}
                >
                  <p
                    className="text secondary disabled"
                    data-component="text"
                    data-typography-element={true}
                  >
                    1 DAY OVERDUE
                  </p>
                </text>
              </div>
            </div>
          </div>
        </div>
      </Cell>
      <Cell
        width="30%"
      >
        <div
          data-component="cell"
          style={
            Object {
              "width": "30%",
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <text
                data-id="taskOwnerName"
                disabled={true}
              >
                <p
                  className="text disabled"
                  data-component="text"
                  data-id="taskOwnerName"
                  data-typography-element={true}
                />
              </text>
            </div>
          </div>
        </div>
      </Cell>
      <Cell
        data-id="manualTask_menu"
        textAlign="center"
        type="ctrlCell"
        width={86}
      >
        <div
          data-component="cell"
          data-id="manualTask_menu"
          style={
            Object {
              "width": 86,
            }
          }
        >
          <div
            className="cell no-padding"
            style={
              Object {
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            <div
              className="cell-inner"
              style={Object {}}
            />
          </div>
        </div>
      </Cell>
    </div>
  </Row>
</TaskCard>
`;

exports[`TaskCard Overdue introduce yourself TaskCard should have a blank icon, use regular text to show task name and have the taskDate highlighted 1`] = `
<TaskCard
  currentUser={
    Object {
      "id": "0e3c5587-b27f-4e2a-bb45-9368cd2727b3",
    }
  }
  formattedValue="Introduce yourself"
  party={
    Object {
      "partyMembers": Array [
        Object {
          "fullName": "Client Name 1",
        },
        Object {
          "fullName": "Client Name 2",
        },
      ],
    }
  }
  selectorData={
    Object {
      "users": Array [
        Object {
          "created_at": "2016-11-19T22:30:39.232Z",
          "fullName": "Alice Altimes",
          "id": "25e5702d-ef29-4996-8948-c1fc593e5b36",
          "preferredName": "Alice",
          "teams": Array [
            Object {
              "displayName": "Parkmerced Pod",
              "id": "d491d706-6129-4edc-a69a-ee5cd026ca68",
              "mainRoles": Array [
                "PM",
              ],
            },
          ],
        },
      ],
    }
  }
  task={
    Object {
      "category": "Party",
      "created_at": "2016-11-23T22:30:39.232Z",
      "dueDate": "2017-05-16T20:00:11+00:00",
      "id": "d5a906a9-5a9b-4e41-a40b-3a57f81d9791",
      "metadata": Object {},
      "name": "INTRODUCE_YOURSELF",
      "partyId": "d5a906a9-5a9b-4e41-a40b-3a57f81d979a",
      "state": "Active",
      "updated_at": "2017-05-18T20:00:11Z",
      "userIds": Array [
        "25e5702d-ef29-4996-8948-c1fc593e5b36",
      ],
    }
  }
>
  <Row
    data-id="appointment-row"
  >
    <div
      className="row"
      data-component="row"
      data-id="appointment-row"
      style={Object {}}
    >
      <Cell
        width={70}
      >
        <div
          data-component="cell"
          style={
            Object {
              "width": 70,
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <Tooltip
                appendToBody={true}
                hoverDelay={300}
                position="top"
                text="INTRODUCE_YOURSELF_TOOLTIP"
              >
                <FlyOut
                  appendToBody={true}
                  closeOnTapAway={true}
                  expandTo="top"
                  hoverDelay={300}
                  useHover={true}
                >
                  <Icon
                    className=""
                    disabled={true}
                    name="checkbox-blank-outline"
                  >
                    <section
                      className="icon disabled"
                      data-component="icon"
                      data-red-icon="true"
                      name="checkbox-blank-outline"
                    >
                      <svg
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "<path d=\\"M19,5v14H5V5h2h3h7H19V3H5C3.9,3,3,3.9,3,5v14c0,1.1,0.9,2,2,2h14c1.1,0,2-0.9,2-2V5c0-1.1-0.9-2-2-2V5z M11,16L11,16L11,16
	L11,16L11,16L11,16L11,16z\\"></path>",
                            }
                          }
                          id="checkbox-blank-outline"
                        />
                      </svg>
                    </section>
                  </Icon>
                </FlyOut>
              </Tooltip>
            </div>
          </div>
        </div>
      </Cell>
      <Cell>
        <div
          data-component="cell"
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <div
                className="taskDetails"
                onClick={[Function]}
              >
                <text
                  disabled={false}
                >
                  <p
                    className="text"
                    data-component="text"
                    data-typography-element={true}
                  >
                    Introduce yourself
                  </p>
                </text>
                <text
                  disabled={false}
                  error={true}
                  secondary={false}
                >
                  <p
                    className="text error"
                    data-component="text"
                    data-typography-element={true}
                  >
                    1 DAY OVERDUE
                  </p>
                </text>
              </div>
            </div>
          </div>
        </div>
      </Cell>
      <Cell
        width="30%"
      >
        <div
          data-component="cell"
          style={
            Object {
              "width": "30%",
            }
          }
        >
          <div
            className="cell"
            style={Object {}}
          >
            <div
              className="cell-inner"
              style={Object {}}
            >
              <text
                data-id="taskOwnerName"
                disabled={false}
              >
                <p
                  className="text"
                  data-component="text"
                  data-id="taskOwnerName"
                  data-typography-element={true}
                />
              </text>
            </div>
          </div>
        </div>
      </Cell>
      <Cell
        data-id="manualTask_menu"
        textAlign="center"
        type="ctrlCell"
        width={86}
      >
        <div
          data-component="cell"
          data-id="manualTask_menu"
          style={
            Object {
              "width": 86,
            }
          }
        >
          <div
            className="cell no-padding"
            style={
              Object {
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            <div
              className="cell-inner"
              style={Object {}}
            />
          </div>
        </div>
      </Cell>
    </div>
  </Row>
</TaskCard>
`;
